db/upgrade.php

```diff
@@ -167,5 +169,24 @@ function xmldb_cms_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2023081401, 'cms');
     }
 
+    if ($oldversion < 2023081600) {
+        $table = new xmldb_table('cms_types');
+        $field = new xmldb_field('idnumber', XMLDB_TYPE_CHAR, 255, null, false, null, '', 'name');
+
+        // Conditionally launch add field.
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        // Give each CMS type a suitably unique ID.
+        $records = cms_types::get_records();
+        foreach ($records as $cmstype) {
+            $cmstype->set('idnumber', uniqid($SITE->shortname, true));
```

Comment:
Wouldn't it be better in this instance to have the `idnumber` field as a concatenation between then name (all lowercase and one word-ed) of the cms_type along with some random digits?

Reply:
The idea is to be globally unique. While uniqid can give sitewide uniqueness, it is the site name that helps to make it globally unique.

Reply:
I suppose as a default that's probably fine since the idea is that a human will manually populate the idnumber field under normal circumstances anyway